File: web/src/api/taskSla.ts
export interface TaskSlaSnapshot {
  policy_key: string;
  policy_name: string;
  status: "normal" | "warning" | "breached" | "completed" | "failed" | string;
  target_seconds: number;
  warning_seconds: number;
  elapsed_seconds: number;
  remaining_seconds: number;
  breach_seconds: number;
  deadline_at: string;
  resolved_at?: string | null;
  resolution_seconds?: number | null;
  target_met?: boolean | null;
}

File: server/app/schemas/task_sla.py
from datetime import datetime

from pydantic import BaseModel


class TaskSlaSnapshot(BaseModel):
    policy_key: str
    policy_name: str
    status: str
    target_seconds: int
    warning_seconds: int
    elapsed_seconds: int
    remaining_seconds: int
    breach_seconds: int
    deadline_at: datetime
    resolved_at: datetime | None = None
    resolution_seconds: int | None = None
    target_met: bool | None = None
